tests/RuleSetTest.php

```diff
@@ -607,4 +654,54 @@ private static function getRuleSetDefinitionsToTestWith()
 
         return $testSet;
     }
+
+    /**
+     * @param string $setName
+     * @param string $ruleName
+     * @param string $targetVersion
+     */
+    private static function assertVersionIsLargestAvailable($setName, $ruleName, $targetVersion)
+    {
+        $maximumPHPUnitVersion = preg_replace('/^@PHPUnit(\d)(\d)Migration:risky$/', '$1.$2', $setName);
+
+        $factory = new FixerFactory();
+        $factory->registerBuiltInFixers();
+        $factory->useRuleSet(new RuleSet([$ruleName => true]));
+        $fixer = current($factory->getFixers());
+
+        foreach ($fixer->getConfigurationDefinition()->getOptions() as $option) {
+            if ('target' === $option->getName()) {
+                $availableVersions = $option->getAllowedValues();
+                $availableVersions = array_diff($availableVersions, ['newest']);
+
+                break;
+            }
+        }
+
+        $availableVersions = array_filter(
+            $availableVersions,
+            function ($version) use ($maximumPHPUnitVersion) {
+                return strcmp($maximumPHPUnitVersion, $version) >= 0;
+            }
+        );
+
+        static::assertTrue(\in_array($targetVersion, $availableVersions, true), sprintf(
+            'Rule "%s" (in rule set "%s") has target "%s", but the rule set is not allowing it (allowed are only "%s")',
+            $fixer->getName(),
+            $setName,
+            $targetVersion,
+            implode('", "', $availableVersions)
+        ));
+
+        rsort($availableVersions);
+        $largestAvailableVersion = reset($availableVersions);
+
+        static::assertSame($largestAvailableVersion, $targetVersion, sprintf(
```

Comment:
the method name is `assertVersionIsLargestAvailable`

are we really testing that version is largest from available ones?

`@PHPUnit50Migration:risky` has `php_unit_dedicate_assert: ['target' => '5.0']`
but `php_unit_dedicate_assert` can be configured for target 5.6.

yet, we don't want to assert that we use largest available target for this rule, as we don't want to target 5.6 in 5.0 ruleset

Reply:
For `@PHPUnit50Migration:risky` version available are only up to `5.0` (according to migration's name), so 5.6 would not be available for this migration.

Maybe different naming, "acceptable" or something?

Reply:
i got what your logic does after reading it 5 times, because the current method name is misleading - claiming it does sth else.

if you have no good name for assertion, at least describe what this assertion does.

Reply:
I've renamed function and variable, hopefully to be more descriptive, what do you think about them now?